### DISCUSSION TITLE
Exputt "Side" putting seems to be more immersive

### DISCUSSION THREAD

**Discussion by jmjava** [springbok/MLM2PRO-GSPro-Connector#68] [General]
I only have a 43" screen on top of a small desk table now but the ultimate plan is to get 85" 4k and put it 2ft off the ground so I can hang the Exputt off the bottom.  I think it will be much more realistic than being 10ft from the screen.  43" is already pretty immersive.  Anyone seeing the same thing?
 
![image](https://github.com/springbok/MLM2PRO-GSPro-Connector/assets/591751/7ca8d2d4-916a-4b14-9e82-aefde5d1aa37)

**Comment by wonder99** [comment]
Looks like a good idea!  I don't have a screen to putt towards, I can only go by the distance to the pin, slope etc, and then putt blindly.  That really sucks for me because in real life, I'm much more of a feel putter.  For the sim, I have to be a whole lot more 'calculated'.

**Reply by jmjava** [reply]
@wonder99  @springbok The mlm2pro connector seems to be the only way to set up Exputt / GSPro for any solution currently.  Would it be possible to get some Developer Setup / documentation so that others can try to help to add / maintain other launch monitors? Even if rapsodo comes out with a native connector I would continue to use as I enjoy Exputt.  I think this is the feature (Exputt) that nobody else has thought of, and it will keep the connector relevant for the the foreseeable future no matter how many native integrations they eventually get.    Standing 3 ft from big screen is way more realistic than 10 ft from projector screen IMHO

**Reply by jmjava** [reply]
Also, have you guys ever thought of integrating with scoring via the API?  Not sure if that is possible.  I have been working on a Flutter app that would allow keeping score for a special type of match called 1 down automatic press.  The score for each match is kept in what ends up looking like a spreadsheet with a new press starting each time somebody wins a hole. I used to be assistant pro and every course I introduced it to, the members instantly adopted it as their game of choice.  Its quite a bit of fun. If data could be pulled from the API on scores it could be pushed out to live DB housed in AWS and that would enable a distributed scoreboard for many matches. Otherwise the scores could be entered via a phone app.

**Reply by springbok** [reply]
Which API, the GSPro API is very limited? It does provide player info, I would have to check exactly what it sends apart from club selection. If this is a 3rd party scoring API it may be possible to send shot data to it for each player.

**Reply by jmjava** [reply]
As for the API, It's just a rough prototype right now with the DB model being generated off a GraphQL schema. https://docs.amplify.aws/flutter/build-a-backend/graphqlapi/data-modeling/  I was able to link it up with a Cognito user pool with only a few lines of code, it's basically their sample app with a portion of my model needed to keep scores for teams and player and to update and persist the "matrix" that represents a series of bets for each match.  What is interesting about Dart/Flutter is its all in one place.. I can really do almost end to end testing persisting back to a serverless DB (DynamoDB) and push out to a grid all from within the same code base. Flutter supports Android, Iphone, Mac Desktop, Linux Desktop, Windows Desktop and the Web from same codebase.

**Comment by springbok** [comment]
> @wonder99 @springbok The mlm2pro connector seems to be the only way to set up Exputt / GSPro for any solution currently. Would it be possible to get some Developer Setup / documentation so that others can try to help to add / maintain other launch monitors? Even if rapsodo comes out with a native connector I would continue to use as I enjoy Exputt. I think this is the feature (Exputt) that nobody else has thought of, and it will keep the connector relevant for the the foreseeable future no matter how many native integrations they eventually get. Standing 3 ft from big screen is way more realistic than 10 ft from projector screen IMHO

I'm looking at integrating the R10 at the moment, I don't own an R10 so adding integration via Bluetooth would be difficult. So I'm looking at allowing the R10 connector to connect to the MLM2PRO connector, the MLM2PRO would send R10 shot data to GSPro. This will allow R10 users to use ExPutt. 

R10<->R10 Connector<->MLM2PRO connector<->GSPro

I originally thought about using OCR of the Garmin App similar to the Rapsodo app but I think using the R10 direct connection would be better. The MML2PRO connector can auto start the R10 connector so should be easy to setup and run, I think it should work well.

**Comment by wonder99** [comment]
Yes, it's quite limited. You could send shot details, but not strokes for
the hole.
Here's the docs

https://gsprogolf.com/GSProConnectV1.html

**Reply by jmjava** [reply]
Yeah so not much integration between what the launcher needs and what the scoring app needs.  I guess it could be used to detect "good" shots and possibly affect the GUI ... or maybe detect bad shots and show bets that would reverse (go from positive to negative) if they continue to play badly for the rest of the hole... Kind of like showing... this will be a 7 bet swing if you lose here... By design 1 down creates many extra presses that each go to end the of the 9.   Any new win / loss affects the running total of all previous bets for each prior "press" .  This is what make it fun... if the GUI could respond to each swing kind of predicting hole outcome like AWS predicts chance of scoring on each play in football... it would be a cool feature.  The original intent was to use phone app for scoring that would send data to live DB and a web page would be used to display match status.  You could have multiple matches displayed at once on that page and would be customized per player.  I am in my house seeing the match from my perspective you are in your house seeing it from your perspective.  When playing in a 4some you can have teams and also play individual matches.  So now... a loss or win affects ALL the running presses in every match you have running.  Having live feedback during a hole might be pretty cool

**Comment by jmjava** [comment]
is the shot number the number of shots so far on the hole?

![image](https://github.com/springbok/MLM2PRO-GSPro-Connector/assets/591751/2e137af2-9abb-4989-9106-4563f65f9460)

**Reply by springbok** [reply]
Shot number is an incrementing number, starts at 1 and gets incremented with every shot.

**Comment by wonder99** [comment]
As far as I know, that shot number isn't used for anything. Probably,
'reserved for future use'.
I think the intention is to distinguish duplicate shots sent close in time.

**Reply by jmjava** [reply]
Ok so the API would have to keep track of current number of strokes just like GSPro does internally then.  That must be doable if they can keep track and we are sending the same data right?  There are other scoring apps out there but I don't think any link up to live sim data and can tap into the AWS analytics type experience

**Reply by jmjava** [reply]
thinking we have raw metrics but we don't know if 300 yds is actually a bad shot... that would be a problem there is no way to know the result I guess

**Reply by jmjava** [reply]
I asked in gspro-api discord channel if adding results to the API would be possible.
